### PR TITLE
[RG-480] Programmer updates: Added program otp, other gui updates.

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
 - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
 - [x] Breaking new feature. If so, please describe details in the description part.

 ### Describe the technical details
Programmer gui updates:
1. Add OPT programming 
image
image

2. Status column: show Fail if process failed. Add text colors 
image

3. Fix for start button: always run all the selected tasks.
4. Disable table context menu when programming is running
5. Checkboxes:
* Make disable by default
* Enable when user select file.
6. Added tool tips

 ### Which part of the code base require a change
 
 - [x] Library: programmer-gui, programmer
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/Configuration/CFGCommon/CFGArg.json [hunk 1]
```diff
@@ -251,4 +251,4 @@
         "  programmer list_cable"
     ]}
   ]
-}
\ No newline at end of file
+}
```
Comment (ravic-rs) [inline src/Configuration/CFGCommon/CFGArg.json:254]: Is this adding whitespace or "}" ?

**Review by ravic-rs — APPROVED** [review]
This looks good, let's wait for @guanyung-rs review too.

File: src/Configuration/CFGCommon/CFGArg.json (hunk 1 above)
Comment (KochynVolodymyr) [reply src/Configuration/CFGCommon/CFGArg.json:254]: it's new line char. QtCreator puts new line char at the end of file by default.

File: src/ProgrammerGui/ProgrammerGuiIntegration.cpp [hunk 2]
```diff
@@ -78,31 +78,40 @@ void ProgrammerGuiIntegration::Devices(const Cable &cable,
...
-  emit this->progress(progress);
+  emit this->progress({m_current.first, m_current.second, m_type}, progress);
 }
 
 void ProgrammerGuiIntegration::ProgramFpga(const Cable &cable,
                                            const Device &device,
                                            const std::string &file) {
   m_current = std::make_pair(cable, device);
-  m_flash = false;
+  m_type = Fpga;
```
Comment (guanyung-rs) [inline src/ProgrammerGui/ProgrammerGuiIntegration.cpp:88]: To be explicit and also consistent to line 105
```suggestion
  m_type = Type::Fpga;
```

File: src/ProgrammerGui/ProgrammerGuiIntegration.cpp [hunk 3]
```diff
@@ -78,31 +78,40 @@ void ProgrammerGuiIntegration::Devices(const Cable &cable,
...
   m_files[device].bitstream = file;
+  emit programStarted({cable, device, m_type});
+}
+
+void ProgrammerGuiIntegration::ProgramOtp(const Cable &cable,
+                                          const Device &device,
+                                          const std::string &file) {
+  m_current = std::make_pair(cable, device);
+  m_type = Otp;
```
Comment (guanyung-rs) [inline src/ProgrammerGui/ProgrammerGuiIntegration.cpp:97]:
```suggestion
  m_type = Type::Otp;
```

**Review by guanyung-rs — APPROVED** [review]
Looks good.

File: src/ProgrammerGui/ProgrammerGuiIntegration.cpp (hunk 2 above)
Comment (KochynVolodymyr) [reply src/ProgrammerGui/ProgrammerGuiIntegration.cpp]: done

File: src/ProgrammerGui/ProgrammerGuiIntegration.cpp (hunk 3 above)
Comment (KochynVolodymyr) [reply src/ProgrammerGui/ProgrammerGuiIntegration.cpp]: done

**Pull request merged**